### ISSUE TITLE
SNR task unite uploaded image

### ISSUE THREAD

**Issue by Lucrezia-Cester** [GSTT-CSC/hazen-web-app#113]
Hi @tomaroberts, 

SNR task requires 2 images as input. Currently when two images are uploaded they are shown separately in the workbench so you need to select both before running the SNR task. This looks slightly unintuitive for the users. 

Here is a solution we could implement:

1) anytime you click on the task to run you see the instructions on the docstring next to the task so that you know how to properly run the task. In the case of SNR the dostring would say to select two images

**Comment by sophie22** [comment]
I agree to have some instructions about the input requirements for specific tasks and I agree that the input files for the SNR task should be linked, and both recorded in the database and in the 'results' output.

As a potential solution, I suggest to add an optional input argument to the SNR task (similar to the ACR SNR task) where the user is prompted to select the image to use for subtraction before SNR calculation. This way it is clear which image was used for which purpose and can be recorded in the database accordingly.

For all task results, I would change the output dictionary to have a section for 1) calculated values, 2) image files used, 3) report images created and an optional 4) additional metadata where any other key-value pairs could be stored.

@tomaroberts and @heyhaleema Let me know what you think.

**Comment by sophie22** [comment]
also see discussion from last year in #75

**Comment by tomaroberts** [comment]
Think we touched upon this in our recent catch-up Sophie, but just adding thoughts here to for the record... 

I think this is inherently tied to the user interface and user experience, so I would be keen to decide how the UI looks and functions for when we have optional arguments (as in the SNR Task). Then we code the backend to fit this. Rather than doing the backend and then crowbarring some UI changes to fit.

**Comment by tomaroberts** [comment]
Very quick mockup, but perhaps something like this:

<img width="650" alt="image" src="https://github.com/GSTT-CSC/hazen-web-app/assets/15610752/45a9d4bc-e752-43c0-9f37-5e76ad779816">

- Dropdown selects the Task.
- Tasks with optional inputs (e.g. SNR) have additional input fields that appear

If we implemented the scrollspy (#52, see [here](https://gstt-csc.github.io/handbook.html#NHS-Staff-Admin/) for working example) (open to other ideas... I'm not 100% convinced is best UI choice), the box above would always be present when you are scrolling through the Studys in the Workbench.
